src/session.rs

```diff
@@ -653,16 +687,50 @@ impl Session {
         let twcc_seq = self.twcc;
         let pad_size = pad_size.map(|p| p.as_bytes_usize());
 
-        if let Some((header, seq_no)) =
+        if let Some((header, seq_no, queued_at)) =
             mline.poll_packet(now, &self.exts, &mut self.twcc, pad_size, buf)
         {
             trace!("Poll RTP: {:?}", header);
+            if let Some(queued_at) = queued_at {
+                {
+                    use std::time::SystemTime;
+
+                    let now = SystemTime::now();
+                    let since_epoch = now.duration_since(SystemTime::UNIX_EPOCH).unwrap();
+                    let unix_time_ms = since_epoch.as_millis();
+                    let delay = Instant::now().duration_since(queued_at).as_millis();
+                    println!("QUEUE_DELAY {},{},{}", header.ssrc, delay, unix_time_ms);
+                }
+            }
+            let kind = if pad_size.is_some() && queued_at.is_none() {
+                "padding"
+            } else {
+                "media"
+            };
+
+            {
+                use std::time::SystemTime;
```

Comment:
This should also be removed before merging, again it's useful though

Reply:
The problem with logging these statistics via the regular logging system is that it introduces a lot of noise which makes parsing more difficult. Ultimately I'm parsing all the stats produced via this and the macros as CSV roughly doing

```python
if line.removeprefix('PACKET_SENT) != line:
  # Had PACKET_SENT prefix
  data = parse_line_as_csv(line.strip(), ['ssrc', 'bits', 'kind', 'timestamp'])
```

Also, the fact that I'm logging to stdout means I can can do

```
cargo run thing > stats.log
```

and still read all the regular logs, which are logged to stderr, without having these, extremely frequent, values clog that up

Reply:
Right. But you can feature flag it, and push it down into `poll_packet` right? And avoid getting that 3rd tuple back from `poll_packet`.